src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -38,6 +30,41 @@ function crosses_vertex(t::Scaled{C,Prod{Op}}, g::AbstractGraph, v) where {C}
   return v ∈ span(t, g)
 end
 
+# map all vertices `w` of `g` except for `v` to the incident edge of `v`
+# which lies in edge_path(g,w,v)
+function map_vertices_to_incident_edges(g::AbstractGraph, v, es)
+  _g = copy(underlying_graph(g))
+  rem_vertex!(_g, v)
+  subgraphs = Set.(connected_components(_g))
+
+  vs = vertextype(g)[]
+  for e in es
+    push!(vs, only(setdiff([src(e), dst(e)], [v])))
+  end
+  _vs = vertextype(g)[]
+  _es = edgetype(g)[]
+
+  for (e, v) in zip(es, vs)
+    for i in eachindex(subgraphs)
+      if v in subgraphs[i]
+        append!(_vs, subgraphs[i])
+        append!(_es, fill(e, length(subgraphs[i])))
+        deleteat!(subgraphs, i)
+        break
+      end
+    end
+  end
+  @assert isempty(subgraphs)
+  return Dict(zip(_vs, _es))
+end
+
+function ordered_incident_edges(g::AbstractGraph, v, ordered_edges)
```

Comment:
Is the issue this is solving that `incident_edges` outputs a set of edges, but they might not be directed in the way you want? It seems a bit arbitrary to put all of that logic into a single function. Conceptually it seems better to split into two functions, like:
```julia
edges = incident_edges(g, v)
aligned_edges = align_edges(edges, reference_edges)
```
and call it like that in the source code. Regarding `align_edges`, here is a conceptually simple implementation:
```julia
julia> using Graphs

julia> edges = [Edge(1 => 2), Edge(2 => 3)]
2-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 1 => 2
 Edge 2 => 3

julia> reference_edges = [Edge(2 => 1), Edge(3 => 2), Edge(4 => 3)]
3-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 2 => 1
 Edge 3 => 2
 Edge 4 => 3

julia> intersect(reference_edges, Iterators.flatten((edges, reverse.(edges))))
2-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 2 => 1
 Edge 3 => 2
```

Reply:
Thanks, that's cleaner! I changed it.